app/code/Magento/Sales/Block/Order/Email/Items/Order/DefaultOrder.php

```diff
@@ -95,10 +96,10 @@ public function getProductAdditionalInformationBlock()
     /**
      * Get the html for item price
      *
-     * @param OrderItem $item
+     * @param OrderItem|CreditMemoItem $item
      * @return string
      */
-    public function getItemPrice(OrderItem $item)
+    public function getItemPrice($item)
```

Comment:
I'd say this is not the best approach to fix it. Do you see any other way to fix it? Through the view models for example.
Looking forward to your response here. 
I know it could take much more time to implement, which is expected, but bringing new bad code to solve existing is not what we want.

Reply:
Agree that's not the best way to solve this. I thought about fixing this in couple of ways

-  new `getItemPrice` method for creditmemo only (this doesn't sound the best way either)
- Using a common Item interface but that's not possible as credit memo item and order item is not related.
- Using $item->getOrderItem() in the creditmemo template but not too sure if it's ideal to do. 

@Stepa4man Do you think any of these solutions are ideal ? If not i'll try to find other way to fix it though it'll take more time to fix it as you mentioned.

Reply:
I noticed here one more thing. This class is marked as API, so it means we can't introduce changes in the method signature, as well as add new methods.
So my suggestion is to copy implementation to the 2 view models with different signatures (). Change phtml to use $viewModel->getItemPrice() and in layout for different cases inject different viewModels

What do you think?

Reply:
I just have one doubt about this approach. Do you think it's feasible to change each an every renderer to do this ? Sorry it's not a so accurate question.

Reply:
Let me clarify my question little bit more. @Stepa4man  

Credit memo renderers use following code bit for item rendering via `sales_email_order_creditmemo_items` 

https://github.com/magento/magento2/blob/fbfac3e51730abe2b79c6072544bc9742fcf7606/app/code/Magento/Sales/Block/Items/AbstractItems.php#L87

The code bit above picks block class respect to product time like : 

- https://github.com/magento/magento2/blob/fbfac3e51730abe2b79c6072544bc9742fcf7606/app/code/Magento/GroupedProduct/view/frontend/layout/sales_email_order_creditmemo_renderers.xml
- https://github.com/magento/magento2/blob/fbfac3e51730abe2b79c6072544bc9742fcf7606/app/code/Magento/Downloadable/view/frontend/layout/sales_email_order_creditmemo_renderers.xml
- https://github.com/magento/magento2/blob/fbfac3e51730abe2b79c6072544bc9742fcf7606/app/code/Magento/Sales/view/frontend/layout/sales_email_order_creditmemo_renderers.xml

We can introduce a ViewModel and started to use new getPrice method from there as you mentioned. 
To achive this we need to change every `sales_email_order_creditmemo_renderers` file and `Magento_Sales::email/items/creditmemo/default.phtml`.


I just wanted to share the operations we need to do so that we can picture the things better.  If you think this way is suitable and better. I can do the changes and test.